extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/MicrometerProcessor.java

```diff
@@ -134,6 +136,18 @@ public List<InterceptorBinding> getAdditionalBindings() {
         return UnremovableBeanBuildItem.beanTypes(METER_REGISTRY, METER_BINDER, METER_FILTER, NAMING_CONVENTION);
     }
 
+    //Not using Optional<TracerProviderBuildItem> that leads to io.quarkus.builder.ChainBuildException
+    @BuildStep
+    void ifOpentelemetryIsEnabeld(CombinedIndexBuildItem indexBuildItem,
```

Comment:
There is a typo here but I really think the method should state what it does, not a condition.
```suggestion
    void ifOpentelemetryIsEnabled(CombinedIndexBuildItem indexBuildItem,
```